src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -135,64 +159,65 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 		}
 	}
 
-	private KafkaCommunicationResult sendAndReceiveMessage() {
+	private void sendMessage() {
 
 		Future<Void> sendReceiveTask = null;
 
 		try {
 
 			sendReceiveTask = executor.submit(() -> {
-				sendAndReceiveKafkaMessage();
+				sendKafkaMessage();
 				return null;
 			});
 
 			sendReceiveTask.get(sendReceiveTimeoutMs, MILLISECONDS);
+			this.kafkaCommunicationResult = KafkaCommunicationResult.success(topic);
```

Comment:
In the `@PostConstruct` method the success and failure results are set in the runnable itself. Why is it done differently here? Can you document why it is different, because this will probably irritate others, too.